lxd/device/cdi/configure.go

```diff
@@ -0,0 +1,339 @@
+package cdi
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sys/unix"
+	"tags.cncf.io/container-device-interface/specs-go"
+
+	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// specDevToNativeDev builds a list of unix-char devices to be created from a CDI spec.
+func specDevToNativeDev(configDevices *ConfigDevices, d specs.DeviceNode) error {
+	if d.Path == "" {
+		return fmt.Errorf("Device path is empty in the CDI device node: %v", d)
+	}
+
+	hostPath := d.HostPath
+	if hostPath == "" {
+		hostPath = d.Path // When the hostPath is empty, the path is the device path in the container.
+	}
+
+	if d.Major == 0 || d.Minor == 0 {
+		stat := unix.Stat_t{}
+		err := unix.Stat(hostPath, &stat)
+		if err != nil {
+			return err
+		}
+
+		d.Major = int64(unix.Major(uint64(stat.Rdev)))
+		d.Minor = int64(unix.Minor(uint64(stat.Rdev)))
+	}
```

Comment:
As a follow-up question: Would it make sense for us to expose the `fillMissingInfo` logic that is used in CDI through a public API: https://github.com/cncf-tags/container-device-interface/blob/main/pkg/cdi/container-edits_unix.go#L60

Reply:
That could be a nice add yes :)

Reply:
Please can an issue be logged to track this @gabrielmougard in GH thanks